File: scores/src/main/java/org/foo/scores/model/Score.java
package org.foo.scores.model;

import io.swagger.v3.oas.annotations.media.Schema;
import org.jspecify.annotations.Nullable;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Schema(description = "Game mode results")
@Table(name="SCORE_BOARD")
public record Score(
        @Schema(readOnly = true)
        @Id Long id,
        Long points,
        String username,
        String mode
) {
}

File: memory-cart-game/src/model/Score.ts
export class Score {
    id?: number;
    points: number;
    username: string;
    mode: string;
    constructor(id: number | null | undefined, points: number, username: string, mode: string) {
        if (id) {
            this.id=id;
        }
        this.points=points;
        this.username=username;
        this.mode=mode;
    }
}